motionplan/tpSpaceRRT.go

```diff
@@ -443,52 +443,56 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
 	rrt rrtMap,
 	nearest node,
 ) (*candidate, error) {
+	// Get the distance function that will find the nearest RRT map node in TP-space of *this* PTG
+	ptgDistOpt, distMap := mp.make2DTPSpaceDistanceOptions(curPtg)
+
 	nm := &neighborManager{nCPU: mp.planOpts.NumThreads / len(mp.tpFrame.PTGSolvers())}
 	nm.parallelNeighbors = 10
 
 	var successNode node
-	// Get the distance function that will find the nearest RRT map node in TP-space of *this* PTG
-	ptgDistOpt := mp.algOpts.distOptions[curPtg]
 
+	var solution *ik.Solution
+	var targetFunc ik.StateMetric
 	if nearest == nil {
 		// Get nearest neighbor to rand config in tree using this PTG
-		// TODO: running nearestNeighbor actually involves a ptg.Solve() call, duplicating work.
 		nearest = nm.nearestNeighbor(ctx, ptgDistOpt, randPosNode, rrt)
 		if nearest == nil {
 			return nil, errNoNeighbors
 		}
+
+		rawVal, ok := distMap.Load(nearest.Pose())
+		if !ok {
+			mp.logger.Error("nearest neighbor failed to find nearest pose in distMap")
+			return nil, errNoNeighbors
+		}
+
+		val, ok := rawVal.(*ik.Solution)
+		if !ok {
+			mp.logger.Error("nearest neighbor ik.Solution type conversion failed")
```

Comment:
My expectation is these logs would be logged twice, once here and once whatever is upstream

Reply:
Sure, but the error returned is the same in all of these branches:
https://github.com/viamrobotics/rdk/pull/3420/files#diff-82a4d2e48ea63f04368d3f5bc9fb16d670e94e76041455dc81d81d9f72b1a35aR460
https://github.com/viamrobotics/rdk/pull/3420/files#diff-82a4d2e48ea63f04368d3f5bc9fb16d670e94e76041455dc81d81d9f72b1a35aR466
https://github.com/viamrobotics/rdk/pull/3420/files#diff-82a4d2e48ea63f04368d3f5bc9fb16d670e94e76041455dc81d81d9f72b1a35aR472

Which makes sense from the end user pespective (as there is not anything they can do to affect the outcome) but if we on the motion team want to be able to diagnose what issue they hit, logging which of these cases restulted in the error is necessary to be able to debug.

Does that make sense? Would you prefer I create new errors for each of these cases?